### PR TITLE
update to extract_animation.py

### PR DESCRIPTION
**Pull request by LegacyID1991**
### Update

The extractor now gets its frame-range from maya animation start and end.
in accordance with #93

### CODE REVIEW

**Comment by LegacyID1991** [comment]
okey, so that would have been to easy.

What should I do here?

`cmds.playbackOptions(query=True, animationEndTime=True)` returns `1`
making the frame-range =
Start 1
End 1

**Comment by LegacyID1991** [comment]
So this is how I solved the test
```
if animstart == animend:
    animend = int(animend) + 100
```
But we probably should try and fix this with a better solution.

File: mindbender/maya/plugins/extract_animation.py [hunk 1]
```diff
@@ -31,6 +31,12 @@ def process(self, instance):
             time=instance.context.data["time"],
             name=instance.data["name"])
 
+        animstart = cmds.playbackOptions(query=True, animationStartTime=True)
+        animend = cmds.playbackOptions(query=True, animationStartTime=True)
```
Comment (mottosso) [inline mindbender/maya/plugins/extract_animation.py]: Try `animationEndTime`.

**Comment by LegacyID1991** [comment]
I corrected my mistake.

But I also had an idea, why no read the context data ?
So in my latest commit the `Start` and `End` is taken from the context collected info,
Is this a bad idea ? 

What do you think ?

**Comment by LegacyID1991** [comment]
And lets ignore that Travis checks failed.

IF the context data read is a good idea then I will think of a way to circumvent that he doesn't have a correct environment.

**Comment by mottosso** [comment]
Just to give you an idea of how it worked before.

1. When an instance is created (via the [Creator GUI](https://github.com/mindbender-studio/core/blob/3e62081449f4e8e6863a37fea0aba932c05744d0/mindbender/tools/creator/app.py#L166), two attributes [are added](https://github.com/mindbender-studio/core/blob/3e62081449f4e8e6863a37fea0aba932c05744d0/mindbender/maya/pipeline.py#L358) to the resulting Maya `objectSet`; [`startFrame` and `endFrame`](https://github.com/mindbender-studio/core/blob/3e62081449f4e8e6863a37fea0aba932c05744d0/mindbender/maya/lib.py#L224).
2. The value these attributes are given is determined by the [registered family](https://github.com/mindbender-studio/core/blob/3e62081449f4e8e6863a37fea0aba932c05744d0/mindbender/maya/pipeline.py#L176).
2. Those attributes were [read by the Collector](https://github.com/mindbender-studio/core/blob/3e62081449f4e8e6863a37fea0aba932c05744d0/mindbender/maya/plugins/collect_instances.py#L84) and stored in the Instance.
3. Finally, the attributes were [used by the Extractor](https://github.com/mindbender-studio/core/blob/3e62081449f4e8e6863a37fea0aba932c05744d0/mindbender/maya/plugins/extract_animation.py#L45).

And hopefully here you can see my mistake. The value given to those attributes was determined at launch, not during the creation of the instance as it should have been.

The reason for doing it this way is to enable you, the developer, to append more customisable attributes to instances that will eventually be editable via a GUI (the Manager). At the moment, these attributes are only accessible via the Attribute Editor in Maya, which makes them a little tricky to notice.

Another reason is for attributes to have a central location and for them to be *reliable*. You should be able to expect certain attributes on a given instance based on its family, so that you can develop plug-ins that *depend* on them.

But we can get back to that once there is a GUI to expose these a little nicer. This approach will work fine for now and solves your immediate problem, which is what is most important. I just wanted you to understand it wasn't this complex by mistake.

> And lets ignore that Travis checks failed.

Let's not. :)

This is what happens when you introduce dependencies in your code (i.e. more required environment variables), and now you'll have to deal with it.

**Comment by LegacyID1991** [comment]
Cool.
Thanks for the info,

> I just wanted you to understand it wasn't this complex by mistake.
 
Ohh no I very took this for a mistake, just didn't see it being used right now, but I understand now that it was there for future implementations and features.
And when we get there it will be AWESOME.

> > And lets ignore that Travis checks failed.

>Let's not. :)

Well I didn't mean forever, ;)  just till I got the answer.

Should I go with the maya read or shall I try and implement a circumvent for Travis CI ?
Maya way is easier..

**Comment by mottosso** [comment]
> Ohh no I never took this for a mistake

It could have been! There certainly *are* mistakes in the code, without a doubt. This bug is an example of that; the value should have been computed during instance creation, but isn't.

I'm certainly not immune to making mistakes, so it's good that you're vigilant.

**Comment by LegacyID1991** [comment]
I went with easy fix for now.

We can always introduce the context data read or when the time is right go with the instance data.

**Comment by mottosso** [comment]
Superseded by #96

**Pull request closed without merging**